src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -303,24 +338,49 @@ void DynamicBuffer::move_data() {
 
     copy(reinterpret_cast<const uint8_t*>(from->GetPtr()), get_ptr(*mem_holder_buffer.get()) + chunk_offset_in_byte,
          src_stride, dst_stride, count, src_stride);
+
+    // adjust for next execution
+    num_iters++;
+    if (map_rule.stride > 0.0f) {
+        chunk_offset_in_byte += mem_holder_unit;
+    } else {
+        chunk_offset_in_byte -= mem_holder_unit;
+    }
 }
 
 void DynamicBuffer::transfer(const Node* node) {
     if (mem_holder_buffer) {
+        const auto axis = map_rule.axis;
+        const auto stride = map_rule.stride;
+        const auto abs_stride = std::abs(stride);
+
+        const auto mem_holder_desc = mem_holder_buffer->get_desc();
+        const auto mem_holder_dims = mem_holder_desc.dims();
+        auto dims = mem_holder_dims;
+
+        dims[axis] = abs_stride * num_iters;
         const auto desc = node->getBaseMemDescAtOutputPort(map_rule.from)->cloneWithNewDims(
-                DnnlExtensionUtils::convertToVectorDims(mem_holder_buffer->get_desc().dims()));
-        redefineToMemories(to, desc);
+                DnnlExtensionUtils::convertToVectorDims(dims));
 
-        copy(get_ptr(*mem_holder_buffer.get()), reinterpret_cast<uint8_t*>(to.front()->GetPtr()), 0, 0, 1, to.front()->GetSize());
+        if (dims[axis] == mem_holder_dims[axis]) {
+            // straightly share mem_holder_buffer to outside graph
+            redefineToMemories(to, desc, mem_holder_buffer->get_data_handle());
+        } else {
+            redefineToMemories(to, desc);  // TODO: redefine only if changed since last inference
```

Comment:
As far as I understand it, `redefineToMemories` itself internally checks whether the shape has changed

Reply:
yes, the comment is a bit nonsense. I will remove it and other commented code.